src/components/elements/table/RestockPurchasesTable.tsx

```diff
@@ -0,0 +1,220 @@
+    },
+    paper: {
+      width: '100%',
+      marginBottom: theme.spacing(2),
+    },
+    table: {
+      minWidth: 750,
+    },
+    visuallyHidden: {
+      border: 0,
+      clip: 'rect(0 0 0 0)',
+      height: 1,
+      margin: -1,
+      overflow: 'hidden',
+      padding: 0,
+      position: 'absolute',
+      top: 20,
+      width: 1,
+    },
+    tableRow: {
+      backgroundColor: '#e3e3e3',
+    },
+    tablePagination: {
+      backgroundColor: '#e3e3e3',
+    },
+  }),
+);
+
+const statusColors: { [key in RestockStatusType]: any } = {
+  completed: 'success',
+  processing: 'info',
+  shipped: 'danger',
+};
+
+type HeadCell = {
+  columnName: keyof RestockDataEntry;
+  label: string;
+  numeric: boolean;
+}
+
+const headCells: Array<HeadCell> = [
+  {columnName: 'id', numeric: false, label: 'Order #'},
+  {columnName: 'purchaseDate', numeric: false, label: 'Purchase Date'},
+  {columnName: 'status', numeric: false, label: 'Status'},
+  {columnName: 'type', numeric: false, label: 'Type'},
+  {columnName: 'totalPrice', numeric: true, label: 'Total Price ($)'},
+];
+
+type EnhancedTableProps = {
+  classes: ReturnType<typeof useStyles>;
+  onRequestSort: (event: React.MouseEvent<HTMLSpanElement>, property: keyof RestockDataEntry) => void;
+  order: Order;
+  orderBy: string;
+}
+
+const EnhancedTableHead = (props: EnhancedTableProps) => {
+  const {classes, order, orderBy, onRequestSort} = props;
+  const handleSort = (property: keyof RestockDataEntry) => (event: React.MouseEvent<HTMLSpanElement>) => {
+    onRequestSort(event, property);
+  };
+
+  const tableClasses = useStyles();
+
+  return (
+    <TableHead>
+      <TableRow className={tableClasses.tableRow}>
+        {headCells.map((headCell) =>
+          <TableCell
+            key={headCell.columnName}
+            sortDirection={orderBy === headCell.columnName ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.columnName}
+              direction={orderBy === headCell.columnName ? order : 'asc'}
+              onClick={handleSort(headCell.columnName)}
+            >
+              {headCell.label}
+              {orderBy === headCell.columnName ? (
+                <span className={classes.visuallyHidden}>
+                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                </span>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>,
+        )}
+        <TableCell>Purchase Info</TableCell>
+      </TableRow>
+    </TableHead>
+  );
+};
+
+const RestockPurchasesTable = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const {restockData} = useRestockSelector();
+
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderBy, setOrderBy] = React.useState<keyof RestockDataEntry>('id');
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(10);
+
+  const getData = () => {
```

Comment:
Instead calling the fetching action via state management, I will just call something like:
```typescript
axios.get().then(dispatch(restockDispatchers.setStockData(someData))));
```

instead.

> Remember to style the code above, I didn't style it.